jobs/payment-jobs/tasks/electronic_funds_transfer_task.py

```diff
@@ -87,23 +90,70 @@ def link_electronic_funds_transfers(cls):
                 continue
 
     @classmethod
-    def _get_eft_short_names_by_state(cls, state: EFTShortnameState) -> List[EFTShortnameModel]:
+    def unlink_electronic_funds_transfers(cls):
+        """Unlink electronic funds transfers."""
+        eft_short_names = cls._get_eft_short_names_by_state(EFTShortnameState.UNLINKED.value)
+        for eft_short_name in eft_short_names:
+            try:
+                current_app.logger.debug(f'Unlinking Electronic Funds Transfer: {eft_short_name.id}')
+                payment_account: PaymentAccountModel = PaymentAccountModel.find_by_auth_account_id(
+                    eft_short_name.auth_account_id)
+                cfs_account: CfsAccountModel = CfsAccountModel.find_effective_by_account_id(payment_account.id)
+                eft_credit: EFTCreditModel = EFTCreditModel.find_by_payment_account_id(payment_account.id)
+
+                payment = db.session.query(PaymentModel) \
+                    .join(PaymentAccountModel, PaymentAccountModel.id == PaymentModel.payment_account_id) \
+                    .join(EFTCreditModel, EFTCreditModel.payment_account_id == PaymentAccountModel.id) \
+                    .join(EFTShortNamesModel, EFTShortNamesModel.id == EFTCreditModel.short_name_id) \
+                    .filter(EFTShortNamesModel.id == eft_short_name.id).first()
+
+                receipt_number = eft_short_name.generate_receipt_number(payment.id)
```

Comment:
When you're generating the receipt number, you're considering the cas_version_suffix?

You can't create duplicate receipt numbers in CAS, just a heads up and you can't use a PUT or PATCH to update the Receipt, unfortunately it has to be recreated.